code/network/multiteamselect.cpp

```diff
@@ -1673,69 +1670,35 @@ void multi_ts_init_objnums()
 	}		
 }
 
-bool multi_ts_validate_ship(char *shipname, char *wingname) 
-{
-	int wing_number, wing_idx; 
-
-	// check name isn't too short to be valid
-	if (strlen(shipname) < (strlen (wingname) + 2) ) {		
-		return false;
-	}
-
-	wing_idx = wing_lookup(wingname); 
-	Assert (wing_idx >= 0 && wing_idx < MAX_WINGS); 
-	wing_number = atoi(shipname+strlen(wingname)); 
-
-	if (wing_number > 0 && wing_number <= Wings[wing_idx].wave_count) {
-		return true; 
-	}
-
-	return false;
-}
-
 // get the proper team and slot index for the given ship name
-void multi_ts_get_team_and_slot(char *ship_name,int *team_index,int *slot_index, bool mantis2757switch)
+void multi_ts_get_team_and_slot(char* ship_name, int* team_index, int* slot_index)
 {
-	int idx; 
 
 	// set the return values to default values
 	*team_index = -1;
 	*slot_index = -1;
 
+	// set up the ship registry pointer
+	const ship_registry_entry* ship_regp = ship_registry_get(ship_name);
```

Comment:
Kinda surprised that clang didn't insist on `auto` here, but that's fine with me.